### PR TITLE
Show ORSP Data Use Limitations

### PR DESCRIPTION
**Pull request by rushtong**
Addresses https://broadinstitute.atlassian.net/browse/GAWB-1760

## Testing
Find a publishable dataset that you can re-publish
Edit the dataset attributes, choose the ORSP-ID option and use some of these as options:
* OD-320 (successful case - shows a text translation)
* OD-330 (error case - not available)
* OD-310 (error case - not found)

There should also be a spinner that shows up while it is retrieving the ORSP information.
You can change back to the questionnaire and the original version of the limitations will be displayed.

---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/summary/library_utils.cljs [hunk 1]
```diff
@@ -48,6 +48,17 @@
...
+  (render-library-row
+    (str "Data Use Limitation for " name)
+    [:span {:style {:whiteSpace "pre-wrap"}} (:translatedUseRestriction consent)]))
+
+(defn render-consent-error [name error]
+  (render-library-row
+    (str "Data Use Limitation not available for " name)
+    (if (:message error)
+      (:message error)
+      error)))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/library_utils.cljs]: You could do:

```clojure
(if-let [message (:message error)]
  message
  error)
```

File: src/cljs/main/broadfcui/page/workspace/summary/library_view.cljs [hunk 2]
```diff
@@ -31,7 +34,21 @@
...
+                                :onClick #(swap! state update :expanded? not)})]]])]))
+   :component-did-mount
+   (fn [{:keys [props state]}]
+     (when-let [orsp-id (:library:orsp (:library-attributes props))]
+       (endpoints/get-consent orsp-id
+         (fn [{:keys [success? get-parsed-response]}]
+           (let [response (get-parsed-response)]
+             (if success?
+               (swap! state assoc :consent (get-parsed-response))
+               (swap! state assoc :consent-error (get-parsed-response))))))))})
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/library_view.cljs]:
```clojure
(fn [{:keys [success? get-parsed-response]}]
  (swap! state assoc (if success? :consent :consent-error) (get-parsed-response)))
```

**Review by MattPutnam — APPROVED** [review]
👍 with two small bits of cleanup

File: src/cljs/main/broadfcui/page/workspace/summary/library_view.cljs (hunk 2 above)
Comment (rushtong) [reply src/cljs/main/broadfcui/page/workspace/summary/library_view.cljs]: Thank you - I was using `let` for debugging the response which is no longer needed!

**Review by davidangb — APPROVED** [review]
This works great! 👍 pending PO approval.

**Comment by davidangb** [comment]
👍 again after latest commit.

File: src/cljs/main/broadfcui/page/workspace/summary/library_utils.cljs [hunk 3]
```diff
@@ -48,6 +48,21 @@
     (get-in library-schema [:properties property-key :title])
     (render-value (get library-attributes property-key))))
 
+(defn render-consent [name consent]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/library_utils.cljs]: Minor nit: `name` is a built-in function, so this is shadowing it.

Comment (rushtong) [reply src/cljs/main/broadfcui/page/workspace/summary/library_utils.cljs]: Thanks - good catch

File: src/cljs/main/broadfcui/page/workspace/summary/library_utils.cljs [hunk 4]
```diff
@@ -48,6 +48,21 @@
...
+(defn render-consent-error [orsp-id error]
+  (render-library-row
+    (str "Structured Data Use Limitations")
+    (let [code (:code error)
+          unapproved (= code 400)
+          not-found (= code 404)]
+      (cond
+        unapproved (str "Structured Data Use Limitations are not approved for " orsp-id)
+        not-found (str "Structured Data Use Limitations are not available for " orsp-id)
+        :else error))))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/library_utils.cljs]: Hmm... I appreciate that the cases are labeled like this, but I'd still prefer:
```clojure
(case (:code error)
  400 (str "Structured Data Use Limitations are not approved for " orsp-id)
  404 (str "Structured Data Use Limitations are not available for " orsp-id)
  error)
```

Comment (rushtong) [reply src/cljs/main/broadfcui/page/workspace/summary/library_utils.cljs]: That does look cleaner.

**Pull request merged**